proving/proving_test.go

```diff
@@ -51,54 +53,60 @@ func TestProver_GenerateProof(t *testing.T) {
 
 	cfg, opts := getTestConfig(t)
 	for numUnits := cfg.MinNumUnits; numUnits < 6; numUnits++ {
-		opts.NumUnits = numUnits
-
-		init, err := NewInitializer(cfg, opts, commitment)
-		r.NoError(err)
-		err = init.Initialize()
-		r.NoError(err)
-
-		p, err := NewProver(cfg, opts.DataDir, commitment)
-		r.NoError(err)
-		p.SetLogger(log)
-
-		binary.BigEndian.PutUint64(ch, uint64(numUnits))
-		proof, proofMetaData, err := p.GenerateProof(ch)
-		r.NoError(err, "numUnits: %d", numUnits)
-		r.NotNil(proof)
-		r.NotNil(proofMetaData)
-
-		r.Equal(commitment, proofMetaData.Commitment)
-		r.Equal(ch, proofMetaData.Challenge)
-		r.Equal(cfg.BitsPerLabel, proofMetaData.BitsPerLabel)
-		r.Equal(cfg.LabelsPerUnit, proofMetaData.LabelsPerUnit)
-		r.Equal(numUnits, proofMetaData.NumUnits)
-		r.Equal(cfg.K1, proofMetaData.K1)
-		r.Equal(cfg.K2, proofMetaData.K2)
-
-		numLabels := cfg.LabelsPerUnit * uint64(numUnits)
-		indexBitSize := uint(shared.BinaryRepresentationMinBits(numLabels))
-		r.Equal(shared.Size(indexBitSize, uint(p.cfg.K2)), uint(len(proof.Indices)))
-
-		log.Info("numLabels: %v, indices size: %v\n", numLabels, len(proof.Indices))
-
-		err = verifying.Verify(proof, proofMetaData)
-		r.NoError(err)
-
-		// Cleanup.
-		err = init.Reset()
-		r.NoError(err)
+		t.Run(fmt.Sprintf("numUnits=%d", numUnits), func(t *testing.T) {
+			opts.NumUnits = numUnits
```

Comment:
It should be fine to run these iterations in parallel, should it not? 🙃 
```suggestion
		t.Run(fmt.Sprintf("numUnits=%d", numUnits), func(t *testing.T) {
			t.Parallel()
			opts.NumUnits = numUnits
```

In general, probably all the tests should be OK to be run in parallel.